lib/saluki-config/src/refresher.rs

```diff
@@ -0,0 +1,90 @@
+use std::sync::Arc;
+
+use crate::GenericConfiguration;
+use reqwest;
+use saluki_error::GenericError;
+use serde::Deserialize;
+use tokio::time::{sleep, Duration};
+
+const DATADOG_AGENT_CONFIG_ENDPOINT: &str = "https://localhost:5004/config/v1/";
+
+#[allow(dead_code)]
+#[derive(Debug, Deserialize)]
+struct ConfigResponse {
+    api_key: String,
+}
+
+/// Configuration for setting up `ConfigRefresher`.
+#[derive(Default, Deserialize)]
+pub struct ConfigRefresherConfiguration {
+    api_key: String,
+    auth_token_file_path: String,
+}
+
+/// The most recent configuration retrieved from the datadog-agent.
+#[derive(Default)]
+pub struct ConfigRefresher {
+    token: String,
+    api_key: String,
```

Comment:
Instead of storing the API key directly, probably just hold something like `serde_json::Value` so that we can query it directly, as well as deserialize the config API response to it.